internal/connector/internal/config/connectors.go

```diff
@@ -119,40 +195,42 @@ func (c *ConnectorsConfig) ReadFiles() error {
 				if os.SameFile(info, prevInfo) {
 					return nil
 				}
-				if typesLoaded[entry.ConnectorType.Id] == path {
+				if typesLoaded[id] == path {
 					return nil
 				}
-				if c.CatalogChecksums[entry.ConnectorType.Id] == sum {
+				if c.CatalogChecksums[id] == sum {
 					return nil
 				}
 
-				return fmt.Errorf("connector type '%s' defined in '%s' and '%s'", entry.ConnectorType.Id, path, prev)
+				return fmt.Errorf("connector type '%s' defined in '%s' and '%s'", id, path, prev)
 			}
 
-			c.CatalogChecksums[entry.ConnectorType.Id] = sum
-			typesLoaded[entry.ConnectorType.Id] = path
+			c.CatalogChecksums[id] = sum
+			typesLoaded[id] = path
 
 			values = append(values, entry)
 
-			glog.Infof("loaded connector %s from file %s", entry.ConnectorType.Id, path)
+			glog.Infof("loaded connector %s from file %s", id, path)
 
 			return nil
 		})
 
 		if err != nil {
 			err = gherrors.Errorf("error listing connector catalogs in %s: %s", dir, err)
-			return err
+			return nil, err
 		}
 	}
 
+	// remove all processed metadata entries
+	for _, entry := range values {
+		delete(connectorMetadata, entry.ConnectorType.Id)
```

Comment:
This is obscure. It would add clarity to return a new list with the remaining ids, rather than modifying the original parameter.

Reply:
The parameter `connectorMetadata` is being merged with the catalog entries. Nothing obscure about it. Maybe you are thinking functional programming. But it's common to modify parameters, see `delete(map,key)`. 
A new list would be sub-optimal use of memory, since neither the original map nor the new list are useful beyond the calling function.

Reply:
It is obscure. There is a syntactical difference from a built in function called `delete` that returns `nil`, and a function called `readConnectorCatalog` returning values. There is no indicator that your function is doing some magic and modifying one of the parameters, I didn't find any indicator and unless I read into the function I don't know what it was doing. If you want to keep it that way it is fine, but returning a new map would add in clarity.

Reply:
I added a comment to explain what the function does with the map.